components/table/src/index.js

```diff
@@ -14,70 +14,77 @@ const TableContainer = styled('table', {
   ({ flexibleColumns }) => ({ tableLayout: flexibleColumns ? 'auto' : 'fixed' }),
 );
 
-const TableHeading = styled('th', {
-  forwardProps: ['name'],
-})(({ rowHeading, columnCount }) => ({
+const cellStyles = {
   ':first-child': {
-    padding: '15px 8px 15px 0',
-    width: rowHeading && columnCount < 4 ? '25%' : undefined,
+    padding: '15px 4px 15px 0',
   },
   ':last-child': {
-    padding: '15px 0 15px 8px',
+    padding: '15px 0 15px 4px',
   },
   borderBottom: '1px solid #bfc1c3',
   display: 'table-cell',
   fontSize: '14px',
-  fontWeight: 'bold',
-  textAlign: 'left',
-  verticalAlign: 'baseline',
-}));
+  padding: '15px 4px',
+};
 
 const TableData = styled('td', {
   // use `forwardProps` here as by default emotion doesn't allow setting `name` prop on a `td`
   forwardProps: ['name'],
-})({
-  ':first-child': {
-    padding: '15px 8px 15px 0',
-  },
-  ':last-child': {
-    padding: '15px 0 15px 8px',
-  },
-  borderBottom: '1px solid #bfc1c3',
-  display: 'table-cell',
-  fontSize: '14px',
-  verticalAlign: 'baseline',
-});
+})(
+  cellStyles,
+  ({ verticalAlign }) => ({ verticalAlign }), 
+);
+
+const TableHeading = styled('th')(
+  cellStyles,
+  ({ verticalAlign }) => ({ verticalAlign }), 
+  ({rowHeading, columnCount}) => (
+    {
+      fontWeight: 'bold',
+      textAlign: 'left',
+      width: rowHeading && columnCount < 4 ? '25%' : undefined,
+    }
+  ),
+);
+
+const getName = (names, row, column, rowIncludesHeading) => {
+  return (rowIncludesHeading) ? names.values[row] : names.values[column];
```

Comment:
changing the mode of operation of `names` so that it's now an object rather than an array is *very* inconvenient - we have dozens of uses of this component that are expecting the array style - we should change the format of names back so that it's expecting a simple array again

Reply:
we may also wish to consider using a different prop than `rowIncludesHeading` to indicate that our use of names should be based on vertical arrangement - it may be the case that we're including headings on rows, but that we still have multiple columns of data that should get differing names based on column number

perhaps this should be a boolean called `nameByRow`

i think that would work very nicely for the majority of cases

for a minority use-case (when `nameByRow` is `false`) the other idea we had of using a 2-d array of names would  allow us to provide explicit names for every single cell on the table.  the usefulness of this however is slightly dubious

Reply:
additionally with the current code this will fail if the `names` object doesn't include a `values` array